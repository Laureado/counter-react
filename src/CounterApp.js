import React, {useState} from 'react'
import PropTypes from 'prop-types'

const CounterApp=({value = 10})=>{

    const [counter, setCounter] = useState(value); 


    //handleSubstract
    const handleSubstract=(number)=>{
        setCounter(counter+number);
        // setCounter((c)=> c+1)
    }

    //handleReset
    const handleReset=()=>setCounter(value);
    
    
    return(
        <>
            <h1>CounterApp</h1>
            <h2>{counter}</h2>

            <button onClick={ ()=> handleSubstract(1) }>+1</button>
            <button onClick={ handleReset }>Reset</button>
            <button onClick={ ()=> handleSubstract(-1) }>-1</button>
        </>
    );
}

CounterApp.propTypes={
    value: PropTypes.number
}

export default CounterApp;