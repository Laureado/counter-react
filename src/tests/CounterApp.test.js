import React from "react";
import "@testing-library/jest-dom";
import { shallow } from "enzyme";

import CounterApp from "../CounterApp";

describe('Pruebas en <CounterApp />', ()=>{

    let wrapper;
    //de este modo abajo te mostrara las referencias que puedes colocar
    // let wrapper = shallow(<CounterApp />);
    //Para que la pagina se renderice otra vez para hacer las pruebas
    beforeEach(()=>{
        wrapper = shallow(<CounterApp />);
    })

    test('Debe mostrar <CounterApp/> Correctamente', ()=>{
        expect(wrapper).toMatchSnapshot();
    })

    test('debe mostrar el valor por defecto de 100', () => {
        const wrapper = shallow(<CounterApp value={100}/>);

        const valorDefault = wrapper.find('h2').text().trim();

        expect(valorDefault).toBe('100');
    })

    test('debe de incrementar con el botón +1', ()=>{
        const btn1 = wrapper.find('button').at(0).simulate('click');

        const valorDefault = wrapper.find('h2').text().trim();

        expect(valorDefault).toBe('11');
    })

    test('debe de decrementar con el botón -1', ()=>{
        
        const btn1 = wrapper.find('button').at(2).simulate('click');
        const valorDefault = wrapper.find('h2').text().trim();
        expect(valorDefault).toBe('9');
    })

    test('debe devolver al valor por defecto', ()=>{
        const wrapper = shallow(<CounterApp value={105}/>);

        const btn1 = wrapper.find('button').at(0).simulate('click');
        const btn2 = wrapper.find('button').at(1).simulate('click');

        const valorDefault = wrapper.find('h2').text().trim();
        expect(valorDefault).toBe('105');
    })
    
})