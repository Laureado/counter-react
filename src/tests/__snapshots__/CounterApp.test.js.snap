// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <CounterApp /> Debe mostrar <CounterApp/> Correctamente 1`] = `ShallowWrapper {}`;
